### DISCUSSION TITLE
How to get list of PRIVATE repositories via api call

### DISCUSSION THREAD

**Discussion by funfunction** [community/community#24382] [Apps, API and Webhooks]
<p>How to get list of PRIVATE repositories via api call</p>
<p>Using this command, it only retrieves the list of PUBLIC repositories…<br>
curl -H “Authorization: token MYTOKEN” <a href="https://api.github.com/users/USERNAME/repos" rel="nofollow noopener">https://api.github.com/users/USERNAME/repos</a></p>
<p>P.S.<br>
My token has all the checkmarks select to enable all permissions, and I still cannot retrieve the list of private repos.</p>
<p>I also tried this and it didn’t work…<br>
curl -H “Authorization: token MYTOKEN” <a href="https://api.github.com/users/USERNAME/repos" rel="nofollow noopener">https://api.github.com/users/USERNAME/repos</a><br>
-d “{“visibility”:“private”}”</p>

**Comment by ilia-cy** [comment, marked as answer]
<p>I’m not sure this is the API you are looking for.<br>
Try this one:</p>
<p><a href="https://api.github.com/search/repositories?q=user:USERNAME" class="onebox" target="_blank" rel="nofollow noopener">https://api.github.com/search/repositories?q=user:USERNAME</a></p>
<p>Without credentials, you will receive all public repositories of the user.<br>
With credentials you will also receive the private ones.</p>

**Reply by vega0** [reply]
what credentials mean. What credentials?

**Reply by omkarsathe01** [reply]
curl -H "Authorization: Bearer YOUR-TOKEN" https://api.github.com/search/repositories?q=user:USERNAME

here replace YOUR-TOKEN with credentials.

**Reply by 010762612** [reply]
😘​

**Reply by vermakhushboo** [reply]
@ilia-cy What if I want to list only the repositories that GitHub app has access to? This means not including all public repos. Only those public repos that the user has given access to the installation.

**Comment by funfunction** [comment]
<p>Thanks, that worked. It retrieves both public and private repos…</p>
<p>curl -H “Authorization: token MYTOKEN” <a href="https://api.github.com/search/repositories?q=user:MYUSERNAME" rel="nofollow noopener">https://api.github.com/search/repositories?q=user:MYUSERNAME</a></p>

**Comment by jonenst** [comment]
<p>I’m looking for an API which returns repos for an owner (a user or an org). In the docs, the search api has<br>
user:USERNAME and org:ORGNAME ( see <a href="https://docs.github.com/en/github/searching-for-information-on-github/searching-for-repositories#search-within-a-users-or-organizations-repositories" rel="nofollow noopener">https://docs.github.com/en/github/searching-for-information-on-github/searching-for-repositories#search-within-a-users-or-organizations-repositories</a> )</p>
<p>In my tests, both return the same results (ie they work for user:ORGNAME and org:USERNAME in addtion to user:USERNAME and org:ORGNAME)</p>
<p>Is this intended ? Is it going to continue to work ?What’s the difference between the two ?</p>
<p>Maybe the keyword “owner” should be used when referring to something that can be a user or an org.</p>

**Comment by jonenst** [comment]
<p>Also<br>
‘<a href="https://api.github.com/users/google/repos?per_page=100&amp;page=11" rel="nofollow noopener">https://api.github.com/users/google/repos?per_page=100&amp;page=11</a>’ works<br>
‘<a href="https://api.github.com/search/repositories?q=user:google&amp;per_page=100&amp;page=11" rel="nofollow noopener">https://api.github.com/search/repositories?q=user:google&amp;per_page=100&amp;page=11</a>’ fails:</p>
<pre><code class="lang-auto">{
  "message": "Only the first 1000 search results are available",
  "documentation_url": "https://docs.github.com/v3/search/"
}
</code></pre>
<p>Could you improve (and  maybe merge?)<br>
<a href="https://api.github.com/users/XXX/repos" rel="nofollow noopener">https://api.github.com/users/XXX/repos</a> or <a href="https://api.github.com/orgs/XXX/repos" rel="nofollow noopener">https://api.github.com/orgs/XXX/repos</a> to return all the results ?<br>
Thanks</p>

**Comment by statusunknown418** [comment]
re opening this, it works with curl but not octokit which seems even weirder, this won't show private repos even while having authorization
```ts
const { data } = await octokit.request("GET /search/repositories", {
        q: `org:${selectedOrganization}`,
        headers: {
          authorization: `Bearer ${token}`,
        },
      });
```

Update:

Looking at the request I can see that gh is not authenticating the user, no f** idea why, getting unauthenticated rate limits
<img width="1054" alt="Screenshot 2023-03-27 at 19 35 36" src="https://user-images.githubusercontent.com/74371045/228096857-3e126719-76a2-4c84-921f-982c72ff5fa8.png">

**Comment by almo8800** [comment]
I couldn`t get private repositories while i was using the wrong link. 
wrong: https:/api.github.com/users/*USERNAME*/repos 
right: https:/api.github.com/user/repos [do not change user to your profile name!!!]

now api give me list of public and private repos

**Reply by notgabry** [reply]
thanks man

**Reply by renettarenula** [reply]
This is correct. 

Though I was using the right endpoint, it was still returning only public repos because I didn't set the scope when [authorizing via oauth](https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/authorizing-oauth-apps#1-request-a-users-github-identity): 

So be sure to set the repo scope if the endpoint is not returning all repositories.

**Reply by amirSohel007** [reply]
it's not working for me.. can somebody help me ? 

`https:/api.github.com/user/repos` Im using this endpoint with get method and passing token in header too.

![Uploading Screenshot 2023-11-23 at 3.32.00 PM.png…]()

**Comment by Rayykoto** [comment]
thanks sir!

**Comment by perfecto25** [comment]
is it possible to list private repos of other users?

Im a GH Enteprise admin at my company, Im creating a python script via GH API that shows all GH users

whenever a user leaves the company, I want to suspend the user, and also transfer his private repos to another organization called "ex-mployees" aka archive his private repos in case someone needs to access them

the current API shows only public repos for a user, unless I'm querying my own username,

```
headers = {'Authorization': f'token {token}'}
response = requests.get(f'{github_url}/users/{user}/repos', headers=headers)
```

not sure if its possible to show private repos for users other than myself.

**Reply by Jon104** [reply]
I'm trying to achieve the same thing, can't figure out how to list other users private repositories i was invited to collaborate...

**Reply by James-Bonner** [reply]
I'm in this situation too.
The [docs](https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#list-repositories-for-the-authenticated-user) imply that 'GET /user/repos' should return all repositories for which a user has accepted an invitation as a collaborator, but it only seems to return ones the user owns or are public. Doesn't even return private org repositories.

**Comment by strikertushar19** [comment]
Hi guys opening this again, i am having difficulty in accessing my private repos i tried  below given links
**https://api.github.com/user/repos?per_page=100&page=$page&visibility=all**
**https://api.github.com/search/repositories?q=user:USERNAME**
Interestingly both are giving different results above one is giving 30 repos and next one 32 repos but but all public 
Even though i tried different versions , Now am confused what else  I have to give . Please share any resources you have..
